ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/UtxoLedgerService.java

```diff
@@ -115,6 +117,46 @@ public interface UtxoLedgerService {
     @Suspendable
     <T extends ContractState> List<StateAndRef<T>> findUnconsumedStatesByExactType(@NotNull Class<T> type);
 
+    /**
+     * Finds unconsumed states of the specified {@link ContractState} type in the vault.
```

Comment:
```suggestion
     * Finds unconsumed states of the specified {@link ContractState} type in the vault. This version supports paging, limiting the number of results returned in a single query call by setting the `limit` argument.
```

I'd also update the doc for the existing call to recommending using this in preference if a (potentially unbounded) number of states may be returned, for performance reasons.